DependencyInjection/DoctrineExtension.php

```diff
@@ -360,10 +360,21 @@ protected function getConnectionOptions(array $connection): array
             throw new InvalidArgumentException('Sharding and master-slave connection cannot be used together');
         }
 
+        foreach (['shards', 'replica', 'slaves'] as $connectionKey) {
+            foreach (array_keys($options[$connectionKey]) as $name) {
```

Comment:
array_intersect_key? Also, I see no reason to do array_keys +$options[$connectionKey][$name]['driverOptions'] ?? [] instead of normal foreach + $connectionOptions[$name]['driverOptions'] ?? []

Reply:
I changed it to foreach without array_keys. Was not sure for what to use `array_intersect_key` tho.